File: app/src/main/java/cn/devifish/readme/provider/BookProvider.java
package cn.devifish.readme.provider;

import cn.devifish.readme.entity.ChapterList;

/**
 * Created by zhang on 2017/3/23.
 *
 * 书籍内容提供者接口
 */

public interface BookProvider {

    /* 获取所有章节列表 */
    public ChapterList getChapterList();

    /* 获取小说详情信息 */
    public String getBookInfo();

    /* 获取小说封面图路径 */
    public String getBookImagePath();

    /* 获取小说更新状态 */
    public String getUpdateInfo();

    /* 获取小说更新时间 */
    public String getUpdateTime();

}

File: app/src/main/java/cn/devifish/readme/provider/BookProvider.kt
package cn.devifish.readme.provider

import cn.devifish.readme.util.Config
import cn.devifish.readme.util.OkHttpUtil
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.fastjson.FastJsonConverterFactory

/**
 * Created by zhang on 2017/7/27.
 *
 */
class BookProvider {

    private object Inner {
        val Instance = Retrofit.Builder()
                .baseUrl(Config.API_BASE_URL)
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .addConverterFactory(FastJsonConverterFactory.create())
                .client(OkHttpUtil.getInstance())
                .build()!!;
    }

    companion object {
        fun getInstance(): Retrofit = Inner.Instance
    }

}